common/src/auth/SharedAccessSignatureBuilder.cs

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Security.Cryptography;
+using System.Text;
+using System.Net;
+
+namespace Microsoft.Azure.Devices.Common.Authorization
+{
+    public class SharedAccessSignatureBuilder
+    {
+        public SharedAccessSignatureBuilder()
+        {
+            TimeToLive = TimeSpan.FromMinutes(20);
+        }
+
+        public string KeyName { get; set; }
+
+        public string Key { get; set; }
+
+        public string HostName { get; set; }
+
+        public TimeSpan TimeToLive { get; set; }
+
+        public virtual string ToSignature()
+        {
+            return BuildSignature(KeyName, Key, HostName, TimeToLive).ToString();
+        }
+
+        protected StringBuilder BuildSignature(string keyName, string key, string Hostname, TimeSpan timeToLive)
```

Comment:
why pass timeToLive as a parameter when the method can access it from the class?

Reply:
Good point.